lib/awful/widget/tasklist.lua

```diff
@@ -332,6 +336,7 @@ local function tasklist_label(c, args, tb)
     local bg_image_minimize = args.bg_image_minimize or theme.tasklist_bg_image_minimize or theme.bg_image_minimize
     local tasklist_disable_icon = args.tasklist_disable_icon or theme.tasklist_disable_icon or false
     local disable_task_name = args.disable_task_name or theme.tasklist_disable_task_name or false
+    local disable_task_name_focus = args.disable_task_name_focus or theme.tasklist_disable_task_name_focus or (disable_task_name and args.disable_task_name_focus == nil and theme.tasklist_disable_task_name_focus == nil)
```

Comment:
It means, if the user sets `disable_task_name = true` and says nothing for `disable_task_name_focus`, it will be sets to `true` as a default?

Shouldn't it be :

```suggestion
    local disable_task_name_focus = args.disable_task_name_focus or theme.tasklist_disable_task_name_focus or false
```

Why do we need to test `disable_task_name` ?

Reply:
My intention was indeed to have i depend on `disable_task_name`.

If `disable_task_name_focus` has `false` as default value, user with `disable_task_name` set to `true` would suddenly see the task name of the focused client, as `disable_task_name_focus` is still set to `false`. 

If this case is acceptable, i. e. the user must explicitly disable the task name of the focused client, i am happy to commit your suggestion, as it would tidy up the code.

Reply:
Sorry, I don't understand the logic...

If we want to preserve the current behavior, and given the changes you did in the test line 407, we should just use `false` as a default here.

Am I missing something? Why do we need to test `disable_task_name` here?

> If disable_task_name_focus has false as default value, user with disable_task_name set to true would suddenly see the task name of the focused client, as disable_task_name_focus is still set to false.

Isn't this the current behavior? (and the one we need to keep?)

Reply:
No problem :)

In the current behaviour, `disable_task_name` set to `true` disables the task name for all clients, regardless of their status (`focused`,`minimized`...)

I implemented the test against `disable_task_name`, that for existing configuration without the newly introduced option, `disable_task_name_focus` would be set to the value of `disable_task_name`.     
With this, the task name of the focused client will NOT been show after the changes by me have been applied.


If, as suggested by your changes, `disable_task_name_focus` would be `false` by default without testing against `disable_task_name`, users with the option `disable_task_name` set to `true` would suddenly see the task-name of the focused client after the changes have been applied.

tl;dr:
`disable_task_name_focus` tests against `disable_task_name`:
`disable_task_name` set to `true` and `disable_task_name_focus` not set => Task name of focused client not shown

`disable_task_name_focus` defaults `false` and doesn't test against `disable_task_name`:
`disable_task_name` set to `true` and `disable_task_name_focus` not set => Task name of focused client will be shown

Hope that the explanation is better understandable now :)

Reply:
Hm... ok! I think those `== nil` got me a little confused here. Thanks for the detailed explanation! :sunglasses:

Reply:
Happy to hear that :)

Reply:
Well, "someone got confused while reading my code" isn't something that one should be happy to hear, especially with regard to maintainability of that code.
Would you be able to summarize that explanation in a code comment?

Reply:
Wrote a short comment, didn't got much into detail, as this would produce clutter in this part of the code.

Also updated the documentation, as the default value for `disable_task_name_focus` is the value of `disable_task_name` instead of `false`.